python/plugins/processing/algs/gdal/rearrange_bands.py

```diff
@@ -0,0 +1,136 @@
+
+    INPUT = 'INPUT'
+    BANDS = 'BANDS'
+    OPTIONS = 'OPTIONS'
+    DATA_TYPE = 'DATA_TYPE'
+    OUTPUT = 'OUTPUT'
+
+    TYPES = ['Byte', 'Int16', 'UInt16', 'UInt32', 'Int32', 'Float32', 'Float64', 'CInt16', 'CInt32', 'CFloat32', 'CFloat64']
+
+    def __init__(self):
+        super().__init__()
+
+    def initAlgorithm(self, config=None):
+        self.addParameter(QgsProcessingParameterRasterLayer(self.INPUT, self.tr('Input layer')))
+        self.addParameter(QgsProcessingParameterBand(self.BANDS,
+                                                     self.tr('Selected band(s)'),
+                                                     None,
+                                                     self.INPUT,
+                                                     allowMultiple=True))
+
+        options_param = QgsProcessingParameterString(self.OPTIONS,
+                                                     self.tr('Additional creation options'),
+                                                     defaultValue='',
+                                                     optional=True)
+        options_param.setFlags(options_param.flags() | QgsProcessingParameterDefinition.FlagAdvanced)
+        options_param.setMetadata({
+            'widget_wrapper': {
+                'class': 'processing.algs.gdal.ui.RasterOptionsWidget.RasterOptionsWidgetWrapper'}})
+        self.addParameter(options_param)
+
+        dataType_param = QgsProcessingParameterEnum(self.DATA_TYPE,
+                                                    self.tr('Output data type'),
+                                                    self.TYPES,
+                                                    allowMultiple=False,
+                                                    defaultValue=5)
+        dataType_param.setFlags(dataType_param.flags() | QgsProcessingParameterDefinition.FlagAdvanced)
+        self.addParameter(dataType_param)
+
+        self.addParameter(QgsProcessingParameterRasterDestination(self.OUTPUT,
+                                                                  self.tr('Converted')))
+
+    def name(self):
+        return 'rearrange_bands'
+
+    def displayName(self):
+        return self.tr('Rearrange bands')
+
+    def group(self):
+        return self.tr('Raster conversion')
+
+    def groupId(self):
+        return 'rasterconversion'
+
+    def icon(self):
+        return QIcon(os.path.join(pluginPath, 'images', 'gdaltools', 'translate.png'))
+
+    def commandName(self):
+        return 'gdal_translate'
+
+    def getConsoleCommands(self, parameters, context, feedback, executing=True):
+        inLayer = self.parameterAsRasterLayer(parameters, self.INPUT, context)
+        if inLayer is None:
+            raise QgsProcessingException(self.invalidRasterError(parameters, self.INPUT))
+
+        out = self.parameterAsOutputLayer(parameters, self.OUTPUT, context)
+
+        arguments = []
+
+        bands = self.parameterAsFields(parameters, self.BANDS, context)
```

Comment:
This looks odd - why would a bands parameter be evaluated by parameterAsFields? 

I think we need a specific parameterAsBands method here.

Reply:
@nyalldawson , yep, the use of parameterAsFields was only temporary until we have that discussion :)

I can see two possible ways forward:
- a new parameterAsStringList, which would essentially be a copy / paste of parameterAsFields
- a new parameterAsBands, which would be a list of int values, and do the "band 1 (blah)" string to int conversion as part of the parameter function

Which method to you prefer?

Reply:
The standard non-multiple band parameter is evaluated using parameterAsInt, so maybe we should keep this general and use something like `parameterAsIntList`?

Reply:
OK, I'll implement that.